sale_commission/wizard/wizard_settle.py

```diff
@@ -140,4 +143,21 @@ def _get_agent_lines(self, agent, date_to_agent):
             ],
             order="invoice_date",
         )
+        if self.date_payment_to:
+            aila = aila.filtered(
+                lambda line: line.commission_id.invoice_state != "paid"
+                or line.invoice_id.invoice_payments_widget == "false"
+                or self.check_payment_date(line)
+            )
```

Comment:
If you want to limit lines to a certain payment date, aren't you with this including all lines that aren't paid as well?

Reply:
I changed it and now I think that's works properly. Is it possible to call check_payment_date method inside domain?

Reply:
unfortunately not. Domains inside a `.search` are translated into SQL and passed to the DB directly. It's also why you can't use fields that are computed and not stored, since they don't exist as columns in the DB.